### DISCUSSION TITLE
Fetch rendered device configuration via API/pynetbox

### DISCUSSION THREAD

**Discussion by jhofmueller** [netbox-community/netbox#13800] [Help Wanted!] [answered]
Hi!

We want to uses the configuration rendering feature that's available now. However I cannot find any hint on how to retrieve the config for a device using the API or (preferably) pynetbox.

**Comment by candlerb** [comment, marked as answer]
Via the REST API documentation: https://demo.netbox.dev/api/schema/swagger-ui/ (or equivalent link on your own Netbox server)

you should find: `POST /api/dcim/devices/{id}/render-config/`

Looking in the source of pynetbox:

```
$ grep -R render.config .
./pynetbox/models/dcim.py:    def render_config(self):
./pynetbox/models/dcim.py:        Represents the ``render-config`` detail endpoint.
./pynetbox/models/dcim.py:        viewing response from the render-config endpoint.
./pynetbox/models/dcim.py:        >>> device.render_config.create()
./pynetbox/models/dcim.py:        return DetailEndpoint(self, "render-config")
```

(Unfortunately, it seems that the DCIM section isn't rendered in the documentation at https://pynetbox.readthedocs.io/ - only IPAM)

**Comment by jhofmueller** [comment]
I must be getting old. Somehow I *lost* the trailing backslash when testing the API call using curl. That obviously doesn't work and I thought something is broken/missing. Sorry for the noise and thanks for your patience :sunglasses:

**Reply by candlerb** [reply]
Glad it's working. `curl -v` would probably have shown a redirect response:

```
> POST /api/dcim/devices/1/render-config HTTP/1.1
...
< HTTP/1.1 301 Moved Permanently
...
< Location: /api/dcim/devices/1/render-config/
```

**Comment by jhofmueller** [comment]
It seems that not only is the pynetbox documentation not rendered but also the functionality of `device.render_config.create()` is still missing. My colleague will file a bug report.

**Reply by candlerb** [reply]
I opened a documentation issue yesterday: https://github.com/netbox-community/pynetbox/issues/583

I did not check the functionality of render_config. It looks like it was added in pynetbox 7.2.0 less than 2 weeks ago, so check you are running the latest version.

**Reply by jhofmueller** [reply]
I can confirm that it works as expected with pynetbox 7.2.0. Thanks again!
